compiler/rustc_hir_analysis/src/check/fn_ctxt/suggestions.rs

```diff
@@ -80,6 +80,38 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let Some((def_id_or_name, output, inputs)) = self.extract_callable_info(expr, found)
             else { return false; };
         if can_satisfy(output) {
+            // suggest `unwrap_or_else` for `Option` and `Result`, when a closure is given
+            if let hir::ExprKind::Closure(..) = expr.kind &&
+                let parent = self.tcx.hir().get_parent_node(expr.hir_id) &&
+                let Some(Node::Expr(call_expr)) = self.tcx.hir().find(parent) &&
+                let hir::ExprKind::MethodCall(
+                      hir::PathSegment {ident: method_name, ..},
+                      receiver,
+                      _args,
+                      _span) = call_expr.kind &&
+                inputs.len() == 0 {
```

Comment:
I don't know how to check the arguments types, currently using `method_exists` to check the function `unwrap_or_else` existed or not.

Reply:
Please check that the `_else` method takes the same number of arguments at least